crates/api/api_utils/src/notify.rs

```diff
@@ -0,0 +1,788 @@
+  PersonNotification::create(&mut context.pool(), &form).await?;
+
+  if data.do_send_email {
+    send_reply_email(
+      &user_view,
+      comment,
+      data.creator,
+      &parent_comment,
+      data.post,
+      context.settings(),
+    )
+    .await?;
+  }
+  Ok(())
+}
+
+async fn notify_mentions(
+  data: &NotifyData<'_>,
+  notif_id: NotificationId,
+  context: &LemmyContext,
+) -> LemmyResult<()> {
+  let mentions = scrape_text_for_mentions(&data.content())
+    .into_iter()
+    .filter(|m| m.is_local(&context.settings().hostname) && m.name.ne(&data.creator.name));
+  for mention in mentions {
+    // Ignore error if user is remote
+    let Ok(user_view) = LocalUserView::read_from_name(&mut context.pool(), &mention.name).await
+    else {
+      continue;
+    };
+
+    let is_blocked = data
+      .check_notifications_allowed(user_view.person.id, context)
+      .await
+      .is_err();
+    if is_blocked {
+      continue;
+    };
+
+    let form =
+      PersonNotificationInsertForm::new(notif_id, user_view.person.id, NotificationTypes::Mention);
+    PersonNotification::create(&mut context.pool(), &form).await?;
+
+    // Send an email to those local users that have notifications on
+    if data.do_send_email {
+      send_mention_email(
+        &user_view,
+        &data.content(),
+        data.creator,
+        data.link(context)?.into(),
+        context.settings(),
+      )
+      .await;
+    }
+  }
+  Ok(())
+}
+
+async fn notify_subscribers(
+  data: &NotifyData<'_>,
+  notif_id: NotificationId,
+  context: &LemmyContext,
+) -> LemmyResult<()> {
+  let subscribers = if data.comment_opt.is_some() {
+    PostActions::list_subscribers(data.post.id, &mut context.pool()).await?
+  } else {
+    CommunityActions::list_subscribers(data.post.community_id, &mut context.pool()).await?
+  };
+
+  for subscriber in subscribers {
+    if data
+      .check_notifications_allowed(subscriber, context)
+      .await
+      .is_err()
+    {
+      continue;
+    };
+
+    let form =
+      PersonNotificationInsertForm::new(notif_id, subscriber, NotificationTypes::Subscribed);
+    PersonNotification::create(&mut context.pool(), &form).await?;
+
+    if data.do_send_email {
+      let user_view = LocalUserView::read_person(&mut context.pool(), subscriber).await?;
+      if let Some(comment) = data.comment_opt {
+        send_post_subscribed_email(
+          &user_view,
+          data.post,
+          comment,
+          data.link(context)?.into(),
+          context.settings(),
+        )
+        .await;
+      } else {
+        send_community_subscribed_email(
+          &user_view,
+          data.post,
+          data.community,
+          data.link(context)?.into(),
+          context.settings(),
+        )
+        .await;
+      }
+    }
+  }
+
+  Ok(())
+}
+
+#[cfg(test)]
+#[expect(clippy::indexing_slicing)]
+#[expect(clippy::unwrap_used)]
+mod tests {
```

Comment:
Moved these tests here from `crates/db_views/inbox_combined` so they can use the notify methods and dont need to insert manually into notification tables (as there are no triggers anymore).